### PR TITLE
Hohmann time to pericenter

### PR DESCRIPTION
**Pull request by jorgepiloto**
Tries to solve #740 and #731 . This pull request includes:

* New orbit class method: `time_to_anomaly`
* Propagates `hohmann` and `bielliptic` to pericenter and adds that `time of flight` in the `Maneuver` instance.
* Makes use of `Quantity.decompose()` to avoid weird behavior with units inside `Maneuver` instances.

### CODE REVIEW

File: src/poliastro/twobody/orbit.py [hunk 1]
```diff
@@ -1024,6 +1024,34 @@ def propagate(self, value, method=mean_motion, rtol=1e-10, **kwargs):
...
+        value : ~astropy.units.Quantity
+
+        Returns
+        -------
+        tof: ~astropy.units.Quantity
+            Time of flight required.
+        """
+
+        # TODO: Avoid repeating logic with mean_motion?
+        p, ecc, inc, raan, argp, _ = rv2coe(
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: Are we using these values anywhere?

File: src/poliastro/maneuver.py [hunk 2]
```diff
@@ -97,8 +97,12 @@ def hohmann(cls, orbit_i, r_f):
             Final altitude of the orbit
         """
 
-        if orbit_i.nu is not 0 * u.deg:
+        # Propagate till periapsis
+        if orbit_i.nu != 0 * u.deg:
+            tof = orbit_i.time_to_anomaly(0 * u.deg)
```
Comment (Juanlu001) [inline src/poliastro/maneuver.py]: Can we add a test for this case?

File: src/poliastro/maneuver.py [hunk 3]
```diff
@@ -184,8 +188,13 @@ def bielliptic(cls, orbit_i, r_b, r_f):
         r_f: astropy.unit.Quantity
             Final altitude of the orbit
         """
-        if orbit_i.nu is not 0 * u.deg:
+
+        # Propagate till periapsis
+        if orbit_i.nu != 0 * u.deg:
+            tof = orbit_i.time_to_anomaly(0 * u.deg)
```
Comment (Juanlu001) [inline src/poliastro/maneuver.py]: Same as above

**Comment by jorgepiloto** [comment]
As commented in the official Riot channel, while working on this, a strange behavior was found in the `get_total_time()` method in the `Maneuver` class. This was caused somehow by the `astropy.units.Quantity`. By making use of the `decompose()` method, this problem was avoided.

This problem was causing #731, so it is also solved :+1:

File: src/poliastro/twobody/orbit.py (hunk 1 above)
Comment (jorgepiloto) [reply src/poliastro/twobody/orbit.py]: Sure! Sorry, did not notice this :sweat:

File: src/poliastro/twobody/orbit.py [hunk 4]
```diff
@@ -1024,6 +1024,27 @@ def propagate(self, value, method=mean_motion, rtol=1e-10, **kwargs):
...
+        Returns
+        -------
+        tof: ~astropy.units.Quantity
+            Time of flight required.
+        """
+
+        # Compute time of flight for correct epoch
+        M = nu_to_M(self.nu, self.ecc)
+        new_M = nu_to_M(value, self.ecc)
+        tof = Angle(new_M - M).wrap_at(360 * u.deg) / self.n
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: Can we reuse this logic in `propagate_to_anomaly`? We could do

```
time_of_flight = self.time_to_anomaly(value)
```

and drop lines 1062 to 1071.

File: src/poliastro/tests/test_maneuver.py [hunk 5]
```diff
@@ -90,3 +137,7 @@ def test_apply_maneuver_correct_dimensions():
 
     assert new_orb.r.ndim == 1
     assert new_orb.v.ndim == 1
+
+
+if __name__ == "__main__":
```
Comment (Juanlu001) [inline src/poliastro/tests/test_maneuver.py]: Leftover from testing, I suppose?

File: src/poliastro/constants.py [hunk 6]
```diff
@@ -286,8 +286,8 @@
 R_earth = Constant(
     "R_earth",
     "Earth equatorial radius",
-    6.3781366e6,
-    "m",
+    6.3781366e3,
```
Comment (Juanlu001) [inline src/poliastro/constants.py]: Can we remove this unrelated change?

Comment (Juanlu001) [reply src/poliastro/constants.py]: Also, remember we still have https://github.com/poliastro/poliastro/issues/731

File: src/poliastro/maneuver.py [hunk 7]
```diff
@@ -141,7 +145,9 @@ def hohmann(cls, orbit_i, r_f):
 
         t_trans = np.pi * np.sqrt(a_trans ** 3 / k)
 
-        return cls((0 * u.s, dv_a), (t_trans, dv_b))
+        return cls(
+            (tof.decompose(), dv_a.decompose()), (t_trans.decompose(), dv_b.decompose())
```
Comment (Juanlu001) [inline src/poliastro/maneuver.py]: Shouldn't this be `t_trans + tof`? (Did not think about it in depth, please confirm)

File: src/poliastro/maneuver.py [hunk 8]
```diff
@@ -234,7 +245,11 @@ def bielliptic(cls, orbit_i, r_b, r_f):
         t_trans1 = np.pi * np.sqrt(a_trans1 ** 3 / k)
         t_trans2 = np.pi * np.sqrt(a_trans2 ** 3 / k)
 
-        return cls((0 * u.s, dv_a), (t_trans1, dv_b), (t_trans2, dv_c))
+        return cls(
+            (tof.decompose(), dv_a.decompose()),
+            (t_trans1.decompose(), dv_b.decompose()),
```
Comment (Juanlu001) [inline src/poliastro/maneuver.py]: Same as above, please confirm we don't have to add `tof` (which shouldn't in fact be called `tof`, but `t_pericenter` or something like that? to avoid confusion) to the other times.

File: src/poliastro/tests/test_maneuver.py [hunk 9]
```diff
@@ -61,6 +61,31 @@ def test_hohmann_maneuver():
     )
 
 
+def test_hohmann_maneuver_not_at_pericenter():
+
```
Comment (Juanlu001) [inline src/poliastro/tests/test_maneuver.py]: Instead of creating a new test, I propose parametrizing `test_hohmann_maneuver`, which is already using the data from example 6-1, and instead of starting from a circular orbit start from an elliptical one, whose pericenter matches the expected altitude. Therefore, when the anomaly is 0, we are in the case where the time to pericenter will be zero, and if we move to a different starting anomaly, we will have a different time.

File: src/poliastro/tests/test_maneuver.py [hunk 10]
```diff
@@ -77,6 +102,28 @@ def test_bielliptic_maneuver():
     assert_quantity_allclose(man.get_total_time(), expected_t_trans, rtol=1e-6)
 
 
+def test_bielliptic_maneuver_not_at_pericenter():
```
Comment (Juanlu001) [inline src/poliastro/tests/test_maneuver.py]: Same comment as above

File: src/poliastro/tests/test_maneuver.py (hunk 5 above)
Comment (jorgepiloto) [reply src/poliastro/tests/test_maneuver.py]: Ups, sorry! :sweat_smile:

File: src/poliastro/twobody/orbit.py (hunk 4 above)
Comment (jorgepiloto) [reply src/poliastro/twobody/orbit.py]: I simplified the `propagate_to_anomaly` by making use of this new method and also using `self.property` instead of calling the `rv2coe` function, which lead to dropping those lines :+1:

File: src/poliastro/constants.py (hunk 6 above)
Comment (jorgepiloto) [reply src/poliastro/constants.py]: Sure! Will open a pull request for this new change :smile:
